phase3/Kconfig

```diff
@@ -37,7 +37,18 @@ config phase3.weave_net
 	bool "Run weave-net?"
 	default n
 	help
-	  Should we deploy weave-net? You likely only want this if using kubeadm for phase2.
+	  [Deprecated] Please use phase3.cni instead.
+
+		Should we deploy weave-net? You likely only want this if using kubeadm for phase2.
+
+config phase3.cni
```

Comment:
It looks like it is possible for a user to apply two network plugins by accident by setting both `weave_net=true` and `cni=flannel` or `cni=weave`.

One solution could be to only present the `phase3.cni` option if `phase3.weave_net==false`

Reply:
I wonder what the versioning model is for this project. Can we just remove `weave_net`?

Reply:
IDK @pipejakob ?

Reply:
Really good question. CCing @mikedanese, the admin of kubernetes-anywhere.

As far as I can tell, kubernetes-anywhere has no versioning of its contents. It doesn't go through any sort of release process, or have a deprecation policy, and only really has the `master` branch.

I was only given maintainer permissions recently, but when I was developing patches for it, I always tried to be a good citizen and maintain backwards-compatibility of the UX as much as possible, although I don't think the project has any general guidelines for it.

@ozdanborne's suggestion to selectively hide the menus sounds like a good idea. It won't protect anyone who is generating `.config` directly (like we do in the e2e tests), but it would at least keep newcomers who are likely using `make menuconfig` from shooting themselves in the foot too much. I've seen too many users turn on every single addon ("why not?") in their configuration, which means trying to install kube-dns and kube-proxy even when using kubeadm, and trying to install weave-net when not using kubeadm, all of which will probably break your cluster. I would go as far as saying we should add even more Kconfig directives to make sure you're not even given the option of weave-net, flannel, cni, etc., unless using kubeadm (although I wouldn't expect that in this PR).